src/midifallback.c

```diff
@@ -0,0 +1,366 @@
+{
+    byte idx;
+    byte checksum;
+
+    // GS allows drums on any channel using SysEx messages.
+    // The message format is F0 followed by:
+    //
+    // 41 10 42 12 40 <ch> 15 <map> <sum> F7
+    //
+    // <ch> is [11-19, 10, 1A-1F] for channels 1-16. Note the position of 10.
+    // <map> is 00-02 for off (normal part), drum map 1, or drum map 2.
+    // <sum> is checksum.
+
+    if (length == 10 &&
+        msg[0] == 0x41 && // Roland
+        msg[1] == 0x10 && // Device ID
+        msg[2] == 0x42 && // GS
+        msg[3] == 0x12 && // DT1
+        msg[4] == 0x40 && // Address MSB
+        msg[6] == 0x15 && // Address LSB
+        msg[9] == 0xF7)   // SysEx EOX
+    {
+        checksum = 128 - ((int)msg[4] + msg[5] + msg[6] + msg[7]) % 128;
+
+        if (msg[8] != checksum)
+        {
+            return;
+        }
+
+        if (msg[5] == 0x10) // Channel 10
+        {
+            idx = 9;
+        }
+        else if (msg[5] < 0x1A) // Channels 1-9
+        {
+            idx = (msg[5] & 0x0F) - 1;
+        }
+        else // Channels 11-16
+        {
+            idx = msg[5] & 0x0F;
+        }
+
+        drum_map[idx] = msg[7];
+    }
+}
+
+static boolean GetProgramFallback(byte idx, byte program,
+                                  midi_fallback_t *fallback)
+{
+    if (drum_map[idx] == 0) // Normal channel
+    {
+        if (bank_msb[idx] == 0 || variation[bank_msb[idx]][program])
+        {
+            // Found a capital or variation for this bank select MSB.
+            selected[idx] = true;
+            return false;
+        }
+
+        fallback->type = FALLBACK_BANK_MSB;
+
+        if (!selected[idx] || bank_msb[idx] > 63)
+        {
+            // Fall to capital when no instrument has (successfully)
+            // selected this variation or if the variation is above 63.
+            fallback->value = 0;
+            return true;
+        }
+
+        // A previous instrument used this variation but it's not
+        // valid for the current instrument. Fall to the next valid
+        // "sub-capital" (next variation that is a multiple of 8).
+        fallback->value = (bank_msb[idx] / 8) * 8;
+        while (fallback->value > 0)
+        {
+            if (variation[fallback->value][program])
+            {
+                break;
+            }
+            fallback->value -= 8;
+        }
+        return true;
+    }
+    else // Drums channel
+    {
+        if (program != drums_table[program])
+        {
+            // Use drum set from drums fallback table.
+            // Drums 0-63 and 127: same as original SC-55 (1.00 - 1.21).
+            // Drums 64-126: standard drum set (0).
+            fallback->type = FALLBACK_DRUMS;
+            fallback->value = drums_table[program];
+            selected[idx] = true;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void MIDI_CheckFallback(midi_event_t *event, midi_fallback_t *fallback)
```

Comment:
`event` should be `const`

Reply:
Fixed in https://github.com/chocolate-doom/chocolate-doom/pull/1558/commits/b4dd44b62276098a2f6ce76e1ee3fd83db2ede34